### DISCUSSION TITLE
Setting `Partition` for auth cookies with `supabase-js`

### DISCUSSION THREAD

**Discussion by tomelliot** [supabase/supabase#22072] [Questions]
I have a widget that runs on a 3rd party domain. The widget is hosted in an iframe. The iframe has a different domain to what's shown in the browser address bar.

Based on [how I understand](https://web.dev/articles/samesite-cookies-explained) cookies to work and the [upcoming Cookies Having Independent Partitioned State changes](https://developers.google.com/privacy-sandbox/3pcd#report-issues), this means I need to set the properties of  supabase's auth cookies to ` SameSite=None; Secure; Partitioned`.

I'm building this widget as a static bundle with Vite, so I'm not using either of the framework helpers or SSR (as [referenced here](https://github.com/orgs/supabase/discussions/3198#discussioncomment-5028280)).

As far as I can see in the docs and the `supabase-js` library, there's no way to set the `Partition` property with `createClient()`. 

What's the right way to handle this with supabase today?

My current implementation (based on the [Next.js auth helper](https://github.com/supabase/auth-helpers/blob/main/packages/nextjs/src/clientComponentClient.ts)) is missing the `Partition` property, and I'm not sure how to include that.

```
import { createClient, SupabaseClient } from "@supabase/supabase-js";
import { Database } from "types/database";
import { BrowserCookieAuthStorageAdapter } from "@supabase/auth-helpers-shared";

const supabaseDeClient: SupabaseClient<Database> = createClient<Database>(
  import.meta.env.VITE_SUPABASE_URL,
  import.meta.env.VITE_SUPABASE_ANON_KEY,
  {
    auth: {
      storage: new BrowserCookieAuthStorageAdapter({
        secure: true,
        sameSite: "none",
        
      }),
    },
  }
);

export default supabaseDeClient;

```

**Comment by tomelliot** [comment]
I've resorted to using localStorage instead of cookies:

```
const supabaseDeClient: SupabaseClient<Database> = createClient<Database>(
  import.meta.env.VITE_SUPABASE_URL,
  import.meta.env.VITE_SUPABASE_ANON_KEY,
  {
    auth: {
      storage: window.localStorage,
    },
  }
);
```

**Comment by vacavaca** [comment]
Plus 1 for that, got the same problem, there are cases when localStorage cannot be used, and the auth cookie must be included with requests to other backend services that use supabase. Setting "Partition" is the only solution I can see to fix the use case with widgets inside iframes.

**Comment by asontha** [comment]
Please fix this!

**Comment by cemalkilic** [comment]
You're using the [deprecated](https://github.com/supabase/auth-helpers?tab=readme-ov-file#supabaseauth-helpers-%EF%B8%8F-deprecated-%EF%B8%8F) `auth-helpers`package, it uses an old version of `cookie` npm package which doesn't support partitioned cookes. As the library was deprecated a while ago, you need to use the [@supabase/ssr](https://github.com/supabase/ssr) package.

You can follow this guide to setup the auth for Next JS: https://supabase.com/docs/guides/auth/server-side/nextjs

While doing the setup, you can pass the `partitioned: true` attribute and cookies would respect it. A basic example:

```javascript
cookiesToSet.forEach(({
   name,
   value,
   options
}) => {
   supabaseResponse.cookies.set({
       name,
       value,
       ...options,
       partitioned: true,
       // You might want to add other options like:
       // secure: true,
   })
})
```
